pkg/metrics/metric/counter.go

```diff
@@ -70,12 +70,28 @@ func NewCounterVec(opts CounterOpts, labelNames []string) DeletableVec[Counter]
 	}
 }
 
+func NewCounterVecWithLabels(opts CounterOpts, labels Labels) DeletableVec[Counter] {
```

Comment:
Seems we have a lot of repetitions here. Perhaps we could compose here aka WithLabels() to decorate the metric vs having to repeat for every metric types?

Reply:
I tried a few approaches here, but the main thing I wanted was to avoid having to declare the labels twice, so any approach besides just passing the labels seemed to yield awkward results.

I did DRY up some of the common code however.